lib/brain.js

```diff
@@ -3180,7 +3143,81 @@ Brain.prototype._physicalDispense = function _physicalDispense () {
   emit('billDispenserDispensing')
 
   const notesToDispense = optimizeDispense(notes, this.billDispenser.dispenseLimit)
-  dispenseGenerator(notesToDispense, this.tx, this.config.completedTimeout)
+  const batchAmount = notesToDispense.length
+
+  const batchDispense = ({ dispensed, current, error }) => {
+    // short circuit out if a error is found
+    if (error) {
+      return { dispensed, error }
+    }
+
+    const notes = notesToDispense[current - 1]
+
+    this.browser().send({ action: 'dispensing', dispenseBatch: { current, of: batchAmount } })
```

Comment:
Not for now, but we should consider replacing this kind of thing with an emitted event, like we do with the board manager. Same will be true of events destined for l-s. Of course, even better would be to have a petri net emit this.